### ISSUE TITLE
Feature: composite build / `includeBuild()` w/ custom init script

### ISSUE THREAD

**Issue by snazy** [gradle/gradle#21143]
TL;DR an option to allow (one or more) init-script(s) for composite builds - analogous to `./gradlew --init-script ...`.

Some scenarios (think: building a comprehensive integration test that includes a couple other projects as included builds) may require custom tweaks to the included builds. Those "tweaks" do not make sense for the included build alone and are very specific to the including build to work - and in many scenarios, developers of the including build does not have "authority" over the included build - or the included build is the (Git) source of an (old) released version (think: `git checkout old-version-1.2.3`).

**Comment by ljacomet** [comment]
Sorry for the late reply!

Why can't this be encoded in the include build? As conditional build logic that executes only when it is effectively included?
Like testing for `Gradle.parent != null`?
